src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -143,6 +144,11 @@ public int Run()
             if (typeSystemContext.InputFilePaths.Count == 0)
                 throw new CommandLineException("No input files specified");
 
+            ilProvider = new HardwareIntrinsicILProvider(
+                instructionSetSupport,
+                new ExternSymbolMappedField(typeSystemContext.GetWellKnownType(WellKnownType.Int32), "g_cpuFeatures"),
```

Comment:
Nit: It looks a bit odd to see this low-level implementation detail in the program Main. I think it would be ok for `HardwareIntrinsicILProvider` to have the `g_cpuFeatures` field name hardcoded. It has the bit masks hardcoded, so it can have the field name hardcoded too.

Reply:
It's actually making things consistent with `g_compilerEmbeddedSettingsBlob` and `g_compilerEmbeddedKnobsBlob` a couple dozen lines below.